lib/bolt/inventory.rb

```diff
@@ -194,27 +200,32 @@ def update_target(target)
     private :update_target
 
     # If target is a group name, expand it to the members of that group.
-    # If a wildcard string, match against nodes in inventory (or error if none found).
-    # Else return [target].
+    # Else match against nodes in inventory by name or alias.
+    # If a wildcard string, error if no matches are found.
+    # Else fall back to [target] if no matches are found.
     def resolve_name(target)
       if (group = @group_lookup[target])
         group.node_names
-      elsif target.include?('*')
+      else
         # Try to wildcard match nodes in inventory
         # Ignore case because hostnames are generally case-insensitive
         regexp = Regexp.new("^#{Regexp.escape(target).gsub('\*', '.*?')}$", Regexp::IGNORECASE)
 
         nodes = []
         @groups.node_names.each do |node|
-          if node =~ regexp
-            nodes << node
-          end
+          nodes << node if node =~ regexp
```

Comment:
can this just be `@groups.node_names.select { |name| name =~ regexp }`

Reply:
Yeah. Next was only slightly more complicated as well.